src/functions/scanOrder/function.ts

```diff
@@ -1,15 +1,15 @@
 import dbContext from '@dbModel/dbContext';
-import Product from '@dbModel/tables/product';
+import Order from '@dbModel/tables/order';
 import { ScanOptions } from '@aws/dynamodb-data-mapper';
 import { objectToConditionExpression } from 'utilities-techsweave';
 
 
-const scanProduct = async (filter: any): Promise<{
-    items: Product[],
-    lastKey: Partial<Product>
+const scanOrder = async (filter: any): Promise<{
+    items: Order[],
+    lastKey: Partial<Order>
 }> => {
-    let items: Product[] = [];
-    let lastKey: Partial<Product>;
+    let items: Order[] = [];
+    let lastKey: Partial<Order>;
```

Comment:
```suggestion
    let lastKey: Partial<Order>;
    let filter = await objectToConditionExpression(filter.filter);
    if (userId) {
        filter = {
            type = 'And',
            conditions: [{
                    type = 'Equals',
                    subject = 'userId',
                    object = userId
                },
                filter 
            ]
    }
```
Maybe you have to switch subject and object, I don't remebember.